crates/atuin/src/command/client/search/interactive.rs

```diff
@@ -1134,12 +1175,53 @@ pub async fn history(
     }
 
     match result {
-        InputAction::Accept(index) if index < results.len() => {
+        InputAction::Accept(index, kind) if index < results.len() => {
             let mut command = results.swap_remove(index).command;
             if accept
                 && (utils::is_zsh() || utils::is_fish() || utils::is_bash() || utils::is_xonsh())
             {
                 command = String::from("__atuin_accept__:") + &command;
+            } else {
+                match kind {
+                    InputAcceptKind::Default => {}
+                    InputAcceptKind::Backspace => {
+                        // trim the end of the selected command *and* remove the
+                        // last character, as tab-completion might have added
+                        // trailing whitespace (which can't be seen in the UI)
+                        command = command.trim_end().to_string();
+                        command.pop();
+                    }
+                    InputAcceptKind::Space => {
+                        // trim the end and add one space character
+                        command = command.trim_end().to_string() + " ";
+                    }
+                    InputAcceptKind::Offset(offset) => {
+                        if settings.exit_positions_cursor {
+                            // for negative offsets (move left), remove trailing whitespace
+                            if offset < 0 {
+                                command = command.trim_end().to_string();
+                            }
+                            #[allow(clippy::cast_possible_wrap)]
+                            let length = command.len() as i64;
+                            let position = if offset >= 0 {
+                                // start editing at specified position,
+                                // counting from the start
+                                length.min(offset)
+                            } else {
+                                // start editing at specified position,
+                                // counting from the (trimmed) end
+                                0.max(length - offset.abs())
+                            };
+                            // Bash's READLINE_POINT is required or positioning the cursor; we still allow
+                            // to go back to the shell even when not using bash (if user's configure the
+                            // respective options), but the actual positioning is only possible with bash
```

Comment:
I think it's possible with Zsh. The text before the cursor position can be stored in `LBUFFER`, and the text after the cursor position can be stored in `RBUFFER`

Reply:
Thanks for the pointer. I'll have a look at L/RBUFFER and zsh.